lib/compiler/src/core_scan.erl

```diff
@@ -357,6 +360,82 @@ name_char($_) -> true;
 name_char($@) -> true;
 name_char(_) -> false.
 
+%% scan_heredocs([Char])
+
+% TODO: Improve error handling
+scan_heredocs(Cs) ->
+    scan_heredocs_1(heredocs_trim_leading(Cs), false, []).
+
+scan_heredocs_1([$",$",$"|Cs], {true, _} = I, Acc) ->
+    {Result, _} =
+        lists:foldl(
+            fun
+            % FIXME: We can have spaces between \ and \n
```

Comment:
This comment makes no sense. The `\` should be the last char before `\n`. I'll remove it.